src/modules/Dashboard/components/TakeNowModal/TakeNowModal.tsx

```diff
@@ -0,0 +1,180 @@
+    const stringSecretIds = joinWihComma(participant.secretIds, true);
+    const id = participant.id ?? participant.details[0].subjectId;
+    let label = stringSecretIds;
+    if (stringNicknames) {
+      label = `${stringSecretIds} (${stringNicknames})`;
+    }
+
+    return { id, label };
+  }, []);
+
+  const { execute: getParticipants } = useAsync(getWorkspaceRespondentsApi, (response) => {
+    if (response?.data) {
+      const data = (response.data as ParticipantsData).result;
+      setParticipants(data.map(participantToOption));
+    }
+  });
+
+  useEffect(() => {
+    if (appletId) {
+      getParticipants({
+        params: {
+          ownerId,
+          appletId,
+          limit: DEFAULT_ROWS_PER_PAGE,
+        },
+      });
+    }
+
+    if (userData) {
+      // For now, the admin is the only member of this list
+      let whoIsResponding = `${userData.user.id}`;
+      if (userData.user.firstName) {
+        whoIsResponding += ` (${userData.user.firstName}`;
+        if (userData.user.lastName) {
+          whoIsResponding += ` ${userData.user.lastName[0].toUpperCase()}.`;
+        }
+        whoIsResponding += ')';
+      }
+
+      const ownerOption = { id: userData.user.id, label: whoIsResponding };
+
+      // In the future we will allow choosing from the full list of participants and team members
+      setParticipantsAndTeamMembers([ownerOption]);
+    }
+  }, [appletId, ownerId, getParticipants, userData]);
+
+  const TakeNowModal = ({ onClose }: TakeNowModalProps) => {
+    const handleClose = () => {
+      setActivity(null);
+      onClose?.();
+    };
+
+    const [subject, setSubject] = useState<AutocompleteOption | null>(defaultSubject);
+    const [participant, setParticipant] = useState<AutocompleteOption | null>(
+      defaultParticipant || participantsAndTeamMembers[0],
+    );
+    const handleSubmit = useCallback(() => {
+      if (subject && participant && activity?.id) {
+        const url = new URL(`protected/applets/${appletId}`, `${process.env.REACT_APP_WEB_URI}/`);
+        url.searchParams.set('startActivityOrFlow', activity.id);
+        url.searchParams.set('subjectId', subject.id);
+        url.searchParams.set('respondentId', participant.id);
+
+        setActivity(null);
+        window.open(url.toString(), '_blank');
+      }
+    }, [subject, participant]);
+
+    if (!activity || !participants || !participantsAndTeamMembers) {
+      return null;
+    }
+
+    return (
+      <Modal
+        open={true}
+        width="57"
+        title={t('takeNow')}
+        buttonText={t('startActivity')}
+        onSubmit={handleSubmit}
+        onClose={handleClose}
+        disabledSubmit={subject === null || participant === null}
+        submitBtnVariant={SubmitBtnVariant.Contained}
+        footerStyles={{ padding: '0.8rem 3.2rem 3.2rem' }}
+        data-testid={`${dataTestId}-take-now-modal`}
+      >
+        <StyledTakeNowModalContent>
+          <StyledTakeNowModalHeader>
```

Comment:
Nit: You could get away with avoiding a `.styles.ts` file just be making use of existing flexbox shared components (I think it's been said that up to 2 style attributes passed via `sx` is fine):

```suggestion
        <StyledFlexColumn sx={{ gap: 0.8, p: theme.spacing(2.4, 3.2) }}>
          <StyledFlexTopCenter sx={{ gap: 2.4 }}>
```

Reply:
Good call 👌